power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -221,35 +216,32 @@ std::vector<Idx2D> get_component_sequence(MainModelState<ComponentContainer> con
 } // namespace detail
 
 template <class... ComponentTypes, class ComponentContainer>
-main_core::utils::SequenceIdx<ComponentTypes...>
+utils::SequenceIdx<ComponentTypes...>
 get_all_sequence_idx_map(MainModelState<ComponentContainer> const& state, ConstDataset const& update_data,
-                         Idx scenario_idx,
-                         main_core::utils::ComponentFlags<ComponentTypes...> const& components_to_store) {
+                         Idx scenario_idx, utils::ComponentFlags<ComponentTypes...> const& components_to_store,
+                         independence::UpdateIndependence<ComponentTypes...> const& independence, bool cached = true) {
     return utils::run_functor_with_all_types_return_array<ComponentTypes...>(
-        [&update_data, &components_to_store, &state, scenario_idx]<typename CompType>() {
-            if (!std::get<main_core::utils::index_of_component<CompType, ComponentTypes...>>(components_to_store)) {
+        [&state, &update_data, scenario_idx, &components_to_store, &independence, cached]<typename CompType>() {
+            auto const comp_properties = std::get<utils::index_of_component<CompType, ComponentTypes...>>(independence);
+            bool const is_comp_independent =
+                cached ? comp_properties.is_independent() : !comp_properties.is_independent();
```

Comment:
I am not very fond of this part of the new implementation... including the `cached = true` argument. However, I couldn't think of anything better. This is the comprimise I made to the the independence calculation outside the logic of this function.

Reply:
yeah that's also why i had the logic with the whole `logical_not` etc. this function shouldn't have a notion as to what "cached" means. the `components_to_store` should already capture everything, right?

Reply:
I am also having trouble understand the logic here. I will have to read more appropriately in the code rather than on github...

Reply:
Tried to improve it a bit by adding a comment, renaming some things, and removing the default of `cached`. Take a look at [c129a50](https://github.com/PowerGridModel/power-grid-model/pull/843/commits/c129a505abde61f497717f07ba7a2114328f1f09).